src/radical/entk/utils/init_transition.py

```diff
@@ -39,3 +39,22 @@ def transition(obj, obj_type, new_state, channel, queue, profiler, logger):
                          logger=logger,
                          local_prof=profiler)
         raise
+
+
+def local_transition(obj, obj_type, new_state, profiler, logger, reporter):
+
+    if obj_type == 'Task':
+        msg = obj.parent_stage['uid']
+    elif obj_type == 'Stage':
+        msg = obj.parent_pipeline['uid']
+    else:
+        msg = None
+
+    obj.state = new_state
+    profiler.prof(  'advance',
+                    uid=obj.uid,
+                    state=obj.state,
+                    msg=msg)
+    reporter.ok('Update: ')
+    reporter.info('%s state: %s\n' %(obj.luid, obj.state))
+    logger.info('Transition of %s to new state %s successful' % (obj.uid, new_state))
```

Comment:
Those reporter messages are very noisy for large numbers of tasks.  But this is unrelated to the PR semantics, just saying :-)

Reply:
True. At that point we can ask users to disable the reporting or have multiple verbosity for reporting as we do for log messages though, right? For now, I am keeping it consistent with Matteo implemented.

Reply:
One could consider a similar mechanism as RP, where state information are less verbose (single character like `+` or `-`).  Or one only reports pipelines and stages, not tasks?

Have a look at the `async_repex` example in `fix/issue_296`: it `reports` pipelines and tasks to `dump.log`.  Do you want me to try to extend that to stages, too, and to pipe that through the reporter?

Reply:
I would be in favor of reporting just pipelines and stages, but at a different reporter level (can we have reporter levels 1,2,3 or so?).

But also, I would like to leave it as is in this PR and revisit this later. Would that be work?